### PR TITLE
posterior : add elbo metric and rename ll to reconstruction_error

### LINKED ISSUE #359: trainer : we should check that early_stopping_metric is in metrics_to_monitor

### LINKED ISSUE #361: trainer : metrics_to_monitor are not monitored, only to_monitor attributes of the posteriors

### PR DESCRIPTION
**Pull request by gabmis**
fixes #361 
fixes #359

### CODE REVIEW

File: scvi/inference/posterior.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def __iter__(self):
 
 class Posterior:
     r"""The functional data unit. A `Posterior` instance is instanciated with a model and a gene_dataset, and
-    as well as additional arguments that for Pytorch's `DataLoader`. A subset of indices can be specified, for
+    as well as additional arguments that for Pytorch"s `DataLoader`. A subset of indices can be specified, for
```
Comment (jeff-regier) [inline scvi/inference/posterior.py]: Did you mean to replace this apostrophe with a quote mark?

File: scvi/inference/posterior.py [hunk 2]
```diff
@@ -46,7 +47,7 @@ class Posterior:
 
     Examples:
 
-    Let's instanciate a `trainer`, with a gene_dataset and a model
+    Let"s instanciate a `trainer`, with a gene_dataset and a model
```
Comment (jeff-regier) [inline scvi/inference/posterior.py]: Here too

File: scvi/inference/fish.py [hunk 3]
```diff
@@ -5,17 +5,24 @@
...
         if verbose:
-            print("LL Fish: %.4f" % ll)
-        return ll
+            print("ELBO Fish: %.4f" % elbo)
+        return elbo
+
+    @torch.no_grad()
+    def reconstruction_error(self, verbose=False):
+        reconstruction_error = compute_reconstruction_error(self.model, self, mode="smFISH")
+        if verbose:
```
Comment (jeff-regier) [inline scvi/inference/fish.py]: Are these `verbose`  flags useful? There's something to be said for just adding in breakpoints for debugging. But I'm just asking.

Comment (jeff-regier) [reply scvi/inference/fish.py]: If they aren't useful, it'd be nice to just directly call `compute_reconstruction_error` rather than adding another method that does almost nothing except call `compute_reconstruction_error`.

File: scvi/inference/fish.py [hunk 4]
```diff
@@ -65,8 +72,8 @@ def __init__(self, model, gene_dataset_seq, gene_dataset_fish, train_size=0.8, t
         self.train_seq, self.test_seq = self.train_test(self.model, gene_dataset_seq, train_size, test_size, seed)
         self.train_fish, self.test_fish = self.train_test(self.model, gene_dataset_fish,
                                                           train_size, test_size, seed, FishPosterior)
-        self.test_seq.to_monitor = ['ll']
-        self.test_fish.to_monitor = ['ll']
+        self.test_seq.to_monitor = ['reconstruction_error']
+        self.test_fish.to_monitor = ['reconstruction_error']
```
Comment (jeff-regier) [inline scvi/inference/fish.py]: Do you want double quotes here? I prefer double quotes I guess but I don't mind a mix either. But other places it seems like you're switching to double.

File: scvi/models/log_likelihood.py [hunk 5]
```diff
@@ -7,12 +7,32 @@
...
-def compute_log_likelihood(vae, posterior, **kwargs):
-    """ Computes log p(x/z), which is the reconstruction error .
-        Differs from the marginal log likelihood, but still gives good
-        insights on the modeling of the data, and is fast to compute
+def compute_elbo(vae, posterior, **kwargs):
+    """ Computes the ELBO.
+
+    The ELBO  is the reconstruction error + the KL divergences
+    between the variational distributions and the priors.
+    Differs from the marginal log likelihood, but still gives good
```
Comment (jeff-regier) [inline scvi/models/log_likelihood.py]: You might say that the ELBO is a lower bound on the marginal log likelihood plus a term that is constant with respect to the variational distribution.

File: scvi/inference/posterior.py (hunk 1 above)
Comment (gabmis) [reply scvi/inference/posterior.py]: Misuse of ctrl-replace I guess

File: scvi/inference/fish.py (hunk 3 above)
Comment (gabmis) [reply scvi/inference/fish.py]: > Are these `verbose` flags useful? There's something to be said for just adding in breakpoints for debugging. But I'm just asking.

Not really, especially especially when we'll have completely switched to logging



> If they aren't useful, it'd be nice to just directly call `compute_reconstruction_error` rather than adding another method that does almost nothing except call `compute_reconstruction_error`.

I did not want to break the calls that had `verbose` param in them, especially in the notebooks since it would more or less require a full re-run to have to corresponding output.

File: scvi/inference/fish.py (hunk 4 above)
Comment (gabmis) [reply scvi/inference/fish.py]: I care for consistency and prefer double quotes. As I was modifying posterior.py I thought i'd harmonize to double quotes. I did not do it here to avoid breaking potential consistency.

File: scvi/models/log_likelihood.py (hunk 5 above)
Comment (gabmis) [reply scvi/models/log_likelihood.py]: Thanks, i'll put that in.

File: scvi/inference/trainer.py [hunk 6]
```diff
@@ -42,8 +42,11 @@ class Trainer:
     default_metrics_to_monitor = []
 
     def __init__(self, model, gene_dataset, use_cuda=True, metrics_to_monitor=None, benchmark=False,
-                 verbose=False, frequency=None, weight_decay=1e-6, early_stopping_kwargs=dict(),
-                 data_loader_kwargs=dict(), show_progbar=True):
+                 verbose=False, frequency=None, weight_decay=0, early_stopping_kwargs=None,
```
Comment (jeff-regier) [inline scvi/inference/trainer.py]: It looks like you're changing `weight_decay` here from 1e-6 to 0. It should be the former I think, right?

**Comment by gabmis** [comment]
@jeff-regier Thank you for your review, this PR is done on my part, feel free to merge.
